onnxruntime/core/providers/cpu/controlflow/loop.cc

```diff
@@ -133,44 +133,44 @@ ONNX_CPU_OPERATOR_KERNEL(Loop,
 
 Loop::Info::Info(const onnxruntime::Node& node, const GraphViewer& subgraph_in)
     : subgraph(subgraph_in) {
-  num_loop_carried_vars = static_cast<int>(node.InputDefs().size()) - 2;  // skip 'M' and 'cond'
-  num_implicit_inputs = static_cast<int>(node.ImplicitInputDefs().size());
+  num_loop_carried_vars = node.InputDefs().size() - 2;  // skip 'M' and 'cond'
+  num_implicit_inputs = node.ImplicitInputDefs().size();
   num_subgraph_inputs = 2 + num_loop_carried_vars;  // iter_num, cond, loop carried vars
-  num_outputs = static_cast<int>(node.OutputDefs().size());
+  num_outputs = node.OutputDefs().size();
 
   // Hold the type for loop carried dependencies - we will use it later
-  const auto& node_input_types = node.InputDefs();
+  const auto &node_input_types = node.InputDefs();
   loop_carried_vars_types.reserve(num_subgraph_inputs);
-  for (int i = 0; i < num_loop_carried_vars; ++i) {
-    loop_carried_vars_types.push_back(node_input_types[static_cast<size_t>(i) + 2]->TypeAsProto());
+  for (uint64_t i = 0; i < num_loop_carried_vars; ++i) {
```

Comment:
Why uint64_t? num_loop_carried_vars seems to have a type of 'size_t'

Reply:
size_t and uint64_t is the same, both are 64 bits, and it has been interchanging used throughout the ort project.